src/main/java/com/linecorp/armeria/client/routing/ServerGroup.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright 2016 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.client.routing;
+
+import java.util.List;
+
+public interface ServerGroup<T extends Node> {
+    List<T> nodes();
+
+    default boolean isReplaceable() {
```

Comment:
For now,maybe replace the whole ServerGroup is more simple way to achieve the same goal.For example,the service's ServerGroup config keep in zookeeper, if some nodes dead,in zookeeper watcher we only read the new config and replace the old groupName registry with  the new ServerGroup.

Implement those interface method also need reconstruct new ServerGroup for target NodeSelector,because the ServerGroup object is immutable.so what do you think?